src/libraries/System.Console/src/System/Console.cs

```diff
@@ -33,31 +33,53 @@ public static class Console
         private static ConsoleCancelEventHandler? s_cancelCallbacks;
         private static ConsolePal.ControlCHandlerRegistrar? s_registrar;
 
-        internal static T EnsureInitialized<T>([NotNull] ref T? field, Func<T> initializer) where T : class =>
-            LazyInitializer.EnsureInitialized(ref field, ref InternalSyncObject, initializer);
+        public static TextReader In
+        {
+            get
+            {
+                return Volatile.Read(ref s_in) ?? EnsureInitialized();
 
-        public static TextReader In => EnsureInitialized(ref s_in, () => ConsolePal.GetOrCreateReader());
+                static TextReader EnsureInitialized()
+                {
+                    ConsolePal.EnsureConsoleInitialized(); // May lock Console.Out
```

Comment:
This avoids the deadlock of https://github.com/dotnet/runtime/issues/34197 which occurs because `EnsureConsoleInitialized` was called under `lock ( s_syncObject)`. By calling it upfront, it no-ops later.